# Artificial Neural Network

### Importing the libraries

In [1]:
import pandas as pd
import numpy as np
import tensorflow as tf

In [2]:
tf.__version__

'2.3.0'

## Part 1 - Data Preprocessing

### Importing the dataset

In [3]:
dataset = pd.read_csv("heart.csv")
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,-1].values

### Splitting the dataset into the Training set and Test set

In [4]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25, random_state = 0)

### Feature Scaling

In [5]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

## Part 2 - Building the ANN

### Initializing the ANN

In [6]:
ann = tf.keras.models.Sequential()

### Adding the input layer and the first hidden layer

In [7]:
ann.add(tf.keras.layers.Dense(units=6, activation="relu"))

### Adding the second hidden layer

In [8]:
ann.add(tf.keras.layers.Dense(units=6, activation="relu"))

### Adding the output layer

In [9]:
ann.add(tf.keras.layers.Dense(units=1, activation="sigmoid"))

## Part 3 - Training the ANN

### Compiling the ANN

In [10]:
ann.compile(optimizer= "adam", loss= "binary_crossentropy" , metrics= ["accuracy"])

### Training the ANN on the Training set

In [11]:
ann.fit(X_train, y_train, batch_size= 32, epochs= 250)

Epoch 1/250
8/8 [==============================] - 0s 1ms/step - loss: 0.7922 - accuracy: 0.5198
Epoch 2/250
8/8 [==============================] - 0s 2ms/step - loss: 0.7617 - accuracy: 0.5198
Epoch 3/250
8/8 [==============================] - 0s 2ms/step - loss: 0.7365 - accuracy: 0.5463
Epoch 4/250
8/8 [==============================] - 0s 1ms/step - loss: 0.7127 - accuracy: 0.5507
Epoch 5/250
8/8 [==============================] - 0s 1ms/step - loss: 0.6917 - accuracy: 0.5815
Epoch 6/250
8/8 [==============================] - 0s 1ms/step - loss: 0.6729 - accuracy: 0.5947
Epoch 7/250
8/8 [==============================] - 0s 2ms/step - loss: 0.6560 - accuracy: 0.5991
Epoch 8/250
8/8 [==============================] - 0s 1ms/step - loss: 0.6399 - accuracy: 0.6123
Epoch 9/250
8/8 [==============================] - 0s 1ms/step - loss: 0.6259 - accuracy: 0.6344
Epoch 10/250
8/8 [==============================] - 0s 2ms/step - loss: 0.6124 - accuracy: 0.6432
Epoch 11/250
8/8 [===========

## Part 4 - Making the predictions and evaluating the model

### Predicting the Test set results

In [12]:
y_pred = ann.predict(X_test)
y_pred = (y_pred > 0.5)

### Making the Confusion Matrix

In [13]:
from sklearn.metrics import confusion_matrix, accuracy_score
cm = confusion_matrix(y_pred, y_test)
print("CM :", cm )
accuracy_score(y_pred, y_test)

CM : [[26  5]
 [ 7 38]]


0.8421052631578947

# Conclusion:

> To evaluate that the patient may have heart disease I have used the dataset of 303 patients and conclude that Artificial Neuron Network is having the accuracy of 84.21 % which is pretty good

> Same dataset I have used in K-Nearest Neighbors model is used and found the accuracy was 81%, where is ANN get the accuracy of 84.21 %